zubhub_frontend/zubhub/src/components/Sidenav/Sidenav.jsx

```diff
@@ -64,24 +69,60 @@ export default function Sidenav() {
         )}
 
         {links({ draftCount, myProjectCount, auth, t }).map(
-          ({ label, link, icon: Icon, red, requireAuth, target }, index) =>
-            displayLink(requireAuth) && (
-              <Link
-                key={index + label}
-                className={clsx(classes.label, classes.link, pathname == link && classes.active, red && classes.red)}
-                href={link}
-                target={target || '_self'}
-              >
-                <ListItem key={label}>
-                  <ListItemIcon>
-                    <Icon size={22} />
-                  </ListItemIcon>
-                  <ListItemText primary={label} />
-                </ListItem>
-              </Link>
-            ),
+          ({ label, link, icon: Icon, red, requireAuth, target, customButton }, index) => {
+            if (displayLink(requireAuth)) {
+              if (customButton) {
+                return (
+                  <Link
+                    key={index + label}
+                    className={clsx(classes.label, classes.link, pathname === link && classes.active, red && classes.red)}
+                    href={link}
+                    target={target || '_self'}
+                  >
+                    <ListItem key={label}>
+                      <ListItemIcon>
+                        <Icon size={22} />
+                      </ListItemIcon>
+                      <ListItemText
+                        primary={
+                          <span>
+                            {label}
+                            <CustomButton
+                              variant="outlined"
+                              customButtonStyle
+                              className={classes.customNumberTag}
+                            >
+                              {label === 'My Drafts' ? (draftCount > 0 ? draftCount : '') : (myProjectCount > 0 ? myProjectCount : '')}
```

Comment:
Do not use labels for comparison, it might fail in non English languages, instead created id, and also use the id for the key